src/k8s-extension/azext_k8s_extension/partner_extensions/OpenServiceMesh.py

```diff
@@ -93,3 +105,58 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
             release_train=release_train,
             version=version
         )
+
+
+def _validate_tested_distro(cmd, cluster_resource_group_name, cluster_name, extension_version):
+
+    if version.parse(str(extension_version)) <= version.parse("0.8.3"):
+        logger.warning('\"testedDistros\" field unavailable for version %s of osm-arc, '
+                       'cannot determine if this kubernetes distribution has been tested '
+                       'for osm-arc', extension_version)
+        return
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Kubernetes' \
+        '/connectedClusters/{2}'.format(subscription_id, cluster_resource_group_name, cluster_name)
+    try:
+        resource = resources.get_by_id(cluster_resource_id, '2020-01-01-preview')
+        cluster_distro = resource.properties['distribution'].lower()
+    except CloudError as ex:
+        raise ex
+
+    if cluster_distro == "general":
+        logger.warning('kubernetes distribution is \"general\", cannot determine if this kubernetes '

```

Comment:
all places that have `osm-arc` should be changed to official name `microsoft.openservicemesh`